### ISSUE TITLE
FTP connection from VSCode 1.34-insider is broken

### ISSUE THREAD

**Issue by gjsjohnmurray** [mkloubert/vscode-remote-workspace#119]
See https://github.com/Microsoft/vscode/issues/72909

When investigating problems my own extension (based on liximomo.remotefs), I tested FTP connections to a couple of sites using mkloubert.vscode-remote-workspace and found that I couldn't even get the tree contents to expand in the following Insiders build:

Version: 1.34.0-insider (user setup)
Commit: 63655183ba5305b70ffaf1327b8a4708f0a79bd9
Date: 2019-04-26T05:21:48.183Z
Electron: 3.1.8
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 10.0.16299

The same VSCode workspace connects OK from 1.33.1.

This line from your log file may be relevant:
```
TRACE fs.ftp.ftpfilesystem.forconnection() - [26/Apr/2019:11:18:53 +0000] "(Error) Expected marks [object Object] instead of: 227 Entering Passive Mode (38,105,105,87,234,130).
```
Perhaps a consequence of a recent VSCode change that means `stat` and `readFile` calls may now run concurrently. See https://github.com/Microsoft/vscode/issues/72891#issuecomment-486966422

**Comment by bpasero** [comment]
@gjsjohnmurray the `readFile` will only ever trigger when you click on a file. Since here we cannot even get the root to expand in the file explorer, there is no `readFile` involved yet.

**Comment by gjsjohnmurray** [comment]
@bpasero - VSCode issues several `stat` and `readFile` calls as soon as a workspace is loaded, looking for .vscode/settings.json and friends.

**Comment by bpasero** [comment]
Yeah that is true, and that changed. As a file system provider you should be aware that you might get called for files that do not exist.
